src/components/UniversalNavbarExpanded/UniversalNavbarExpanded.js

```diff
@@ -0,0 +1,349 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import uuidv4 from 'uuid/v4'
+import FancyAnimatedLogo from '../UniversalNavbar/FancyAnimatedLogo'
+import LogoNotAnimated from '../UniversalNavbar/assets/ethos-logo-black.js'
+import LogoWhite from '../UniversalNavbar/assets/ethos-logo-white.js'
+import {
+  AccountIcon,
+  SearchIcon,
+  AccordionToggleIcon,
+} from '../UniversalNavbar/assets/icons.js'
+import {
+  Body,
+  Button,
+  Layout,
+  Spacer,
+  TitleMedium,
+  TitleSmall,
+  Footnote,
+  COLORS,
+} from '../index'
+import TransformingBurgerButton from '../UniversalNavbar/TransformingBurgerButton/TransformingBurgerButton'
+import { LINKS } from './constants.js'
+import NavLink from '../UniversalNavbar/NavLink'
+import styles from './UniversalNavbarExpanded.module.scss'
+
+// TODO: convert from class to hook?
+class UniversalNavbarExpanded extends React.Component {
+  state = {
+    showMobileMenu: false,
+    activeAccordionItem: false,
+  }
+
+  toggleHamburger = () => {
+    this.setState({ ...this.state, showMobileMenu: !this.state.showMobileMenu })
+  }
+
+  handleHamburgerKeyPress = (event) => {
+    if (event.key === 'Enter') {
+      this.toggleHamburger(true)
+    }
+  }
+
+  toggleAccordionItem = (toggledItem) => {
+    this.setState({
+      ...this.state,
+      activeAccordionItem:
+        this.state.activeAccordionItem === toggledItem ? false : toggledItem,
+    })
+  }
+
+  handleAccordionItemKeyPress = (event, index) => {
+    if (event.key === 'Enter' || event.which == 13 || event.keyCode == 13) {
+      this.toggleAccordionItem(index)
+    }
+  }
+
+  render() {
+    const {
+      LinkComponent,
+      hideMobileCta,
+      hideDesktopCta,
+      logoHref,
+    } = this.props
+
+    const renderCtaButton = (showWhenScrolled) => (
+      <a
+        className={
+          'cta-button ' + (showWhenScrolled ? 'show-when-scrolled' : '')
+        }
+        onClick={this.props.trackCtaClick}
+        href={LINKS.TERM.href}
+      >
+        <Button.Small.BlackOutline>Check my price</Button.Small.BlackOutline>
+      </a>
+    )
+
+    const { showMobileMenu, activeAccordionItem } = this.state
+    console.log(this.state)
+    const renderTextLink = (link) => (
+      <div key={link.id} className={styles.textLink}>
+        <NavLink
+          href={link.href}
+          LinkComponent={link.href !== '/login/' ? LinkComponent : null}
```

Comment:
Can we just make this a separate constant / function / component above, rather than passing in `link.href !== '/login/' ? LinkComponent : null` to every `LinkComponent` prop

Reply:
yea

Reply:
marked in TODOs for fastfollow